### DISCUSSION TITLE
"Randomise" sorting in browse

### DISCUSSION THREAD

**Discussion by mirutoyoshi** [jcorporation/myMPD#457] [Ideas]
Hi jc,
I think it could be good to be able to get a randomised view of the covergrid results. Perhaps by a sort option.

I find sometimes I don't have a plan of what to put on, so I browse, but always start with the same albums in the browser - ABC or ZYX etc...
Mixing it up would be great to get some inspiration!

**Comment by jcorporation** [comment]
I like the idea of exploring albums randomly, but I am unsure if this is a good way.

**Reply by mirutoyoshi** [reply]
Great, hope the idea goes somewhere.
For sure, the way I suggested may be a bad way to implement it :grin:
But I would love to see a way to present a random grid of albums to explore in a release someday.

**Reply by jcorporation** [reply]
Do you know an other player with this feature? 

For a random queue you can use the jukebox.

**Reply by mirutoyoshi** [reply]
No, I don't any players with a covergrid / music browser that do that. There are players that will shuffle by album in a playlist, but that's more like Jukebox mode (which is a very nice feature, but a different one). Websites - youtube, spotify, pandora suggest things (not randomly) in the music browser, but they have huge libraries and use a lot of models based on user data to make those suggestions. I think this would be difficult to work in a personal music player without some online component.

I think a better example are photo library apps, where the landing page is a random selection from the photostream. Photostructure does this as an example. Others apps present older images based on some other criteria - e.g. 'three years ago today'. But the point is that you see something a bit unexpected and you might go explore the library from that 'new' entry point. Now most people's photo libraries are probably bigger than their music collection, and will be less well tagged with metadata, so it could be argued that the 'random' page approach works for photos because of that...

**Reply by jcorporation** [reply]
Hm, an explore view as an option could be an idea. This view can display albums or songs randomly and/or by some criteria. We can brainstorm here, which criteria could make sense.

Integration of listenbrainz could help to get suggestions.

**Reply by john-** [reply]
I put a bunch of random music into the queue with the jukebox function and then delete the ones I don't want to listen to :)

**Reply by mirutoyoshi** [reply]
A few ideas...

If it was just based on mpd internal data, rather than MusicBrainz/ListenBrainz etc, then some tags that make sense to me for filtering the pool to draw randomly from would be:
- Genre  (if the collection is well tagged in that regard - mine's not a great example)
- Date (e.g. explore mid-90's music)
- Playcount (show a mix of often played and infrequently played)

If you can use the "[tracks you might like](https://community.metabrainz.org/t/tracks-that-you-might-like/495812 )" ListenBrainz feature that would be cool. I'm not sure how you restrict that response to albums in your library though.

Similar to the Listenbrainz feature, if you can pull in aditional metadata from Musicbrainz or similar service, then I think something like this would be neat:
1. Show the initial 'random' selection
2. Click on a particular album
3. Get data from "Relationships", "Aliases","tags" etc from Musicbrainz. Maybe even search again on the related artists to get their other bands. 
4. Search the MPD library for other albums that match the gathered data and present them

But that's probably a lot of work, and could could give bad user experience if it is unresponsive in step 3 or 4... It probably would need a local database to be built, or be done at the server side. Either way, it sounds like a different tool/project to mympd!

**Reply by jcorporation** [reply]
Listenbrainz and musicbrainz integration is on the roadmap. I am willing to invest time to enrich the metadata and provide a cool explore feature.

A local database is out of scope of myMPD, we should fetch all data online from musicbrainz. This should be done on server side, so we evtl. can cache some results.

**Comment by jcorporation** [comment]
myMPD v18 adds home widgets that can display random albums.

**Reply by bhartshorn** [reply]
Hi, I'm curious how to set this up, I'm looking around at making shortcuts and adding widgets but don't (yet) see the way. Is it documented somewhere?

FWIW, Ultrasonic - a Subsonic client for Android, has a Random sort in the album view and I love it. Like the OP, I just want to browse my music looking for something a little different and viewing the albums in random order easily facilitates that.

**Reply by jcorporation** [reply]
First import the script WidgetAlbums and then add the script widget to the homepage.
